nose2/plugins/result.py

```diff
@@ -90,6 +90,15 @@ def testOutcome(self, event):
             else:
                 self.reportCategories['unexpectedSuccesses'].append(event)
                 self._reportUnexpectedSuccess(event)
+        elif event.outcome == result.SUBTEST:
+            # subtests do not report success
+            if event.exc_info is not None:
+                if issubclass(event.exc_info[0], event.test.failureException):
+                    self.reportCategories['failures'].append(event)
+                    self._reportFailure(event)
+                else:
+                    self.reportCategories['errors'].append(event)
+                    self._reportError(event)
```

Comment:
This block handles subtest results. It directly corresponds to the code in [unittest.TestCase](https://github.com/python/cpython/blob/b745a6143ae79efe00aa46affe5ea31a06b0b532/Lib/unittest/case.py#L514).